File: app/src/main/java/com/yu/functionbox/event/EventMessage.java
package com.yu.functionbox.event;

import java.util.List;

/**
 *
 * Created by admin on 2016/6/29.
 */
public class EventMessage<T> {

    public T mObj;
    public List<T> mList;
    public int mCode;

    public EventMessage(int code, List<T> list){
        this.mList = list;
        this.mCode = code;
    }

    public EventMessage(int code) {
        mCode = code;
    }

    public EventMessage(int code, T obj) {
        mCode = code;
        this.mObj = obj;
    }

}

File: app/src/main/java/com/yu/functionbox/event/EventMessage.kt
package com.yu.functionbox.event

/**
 *
 * Created by admin on 2016/6/29.
 */
class EventMessage<T> {

    var mObj: T ?= null
    var mList: List<T> ?= null
    var mCode: Int = 0

    constructor(code: Int, list: List<T>) {
        this.mList = list
        this.mCode = code
    }

    constructor(code: Int) {
        mCode = code
    }

    constructor(code: Int, obj: T) {
        mCode = code
        this.mObj = obj
    }

}
